### PR TITLE
Ensure CHaP builds with the released version of consensus

### PR DESCRIPTION
**Pull request by Jasagredo**
# Description

Ensures CHaP will be able to build the newly released version before telling the user to push the updated version.

### CODE REVIEW

File: scripts/check-consensus-builds-with-chap.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+
+# remove network packages from the list of local packages
+
+sed 's/\.\/.*network.*//g' cabal.project \
```
Comment (nfrisby) [inline scripts/check-consensus-builds-with-chap.sh:7]: This `sed` line seems practical, but it also seems like something that will often cause confusion.

For example, I think a better solution would be:

- enumerate the packages we want to exclude
- enumerate the packages we want to include
- fail if the packages we find in the actual file are not equal the union of the above
- generate the new file by removing the ones we want to exclude

File: scripts/consensus-release.sh [hunk 2]
```diff
@@ -258,6 +258,16 @@ else
...
-echo "Succesfully created release. You should now inspect the current branch (${branch%?}) and if everything looks right, push it to GitHub, open a PR and get it merged!"
+echo "Checking that CHaP would be able to build these packages (this will take a while!)"
+./scripts/check-consensus-builds-with-chap.sh
+
+if [[ -z $? ]]; then
+  echo "Succesfully created release. You should now inspect the current branch (${branch%?}) and if everything looks right, push it to GitHub, open a PR and get it merged!"
+else
+  echo "Your release seem to depend on changes that are not visible in CHaP/Hackage"
+  echo "Maybe you added a function to a separate package or are using a package that is not in those indices?"
+  echo "I created the branch ${branch%?} but it is probably wrong, so you should delete it, check the error above and re-run this script. I will leave the branch just in case."
```
Comment (nfrisby) [inline scripts/consensus-release.sh:272]: Is this line a copy-paste error?

Comment (Jasagredo) [reply scripts/consensus-release.sh:272]: I don't know specifically what you mean. But my intention was to get that message on the screen, having the script talking as "I" :smile:

File: scripts/check-consensus-builds-with-chap.sh (hunk 1 above)
Comment (Jasagredo) [reply scripts/check-consensus-builds-with-chap.sh:7]: Reasonable, let me rework it

Comment (amesgen) [reply scripts/check-consensus-builds-with-chap.sh:7]: Another alternative: move the `packages:` part to two standalone files (one for network, one for consensus) and [import them](https://cabal.readthedocs.io/en/latest/cabal-project.html#conditionals-and-imports), then you can replace the network file with an empty file here temporarily.

File: scripts/consensus-release.sh (hunk 2 above)
Comment (nfrisby) [reply scripts/consensus-release.sh:272]: Ah, OK. That seemed odd to me in the PR diff, but I don't think it would confuse me on the command-line output.

**Review by dnadales — APPROVED** [review]

**Pull request merged**